### ISSUE TITLE
Confirm all version numbers and IDs are ready for production

### ISSUE THREAD

**Issue by mikebattista** [microsoft/TemplateStudio#4461]
### Describe the bug

I've seen various version numbers floating around:

* wizard version = 1.0 (wizard and TemplateStudio.xml)
* templates version = 2.1.0 (TemplateStudio.xml)
* assembly version = 5.x.0 (*.vstemplate)

Let's synchronize them if we need to and also audit the extension IDs to make sure they're what we want.

### Applies to the following platforms:

- [X] WinUI
- [X] WPF
- [X] UWP

### About your setup

 - Visual Studio Version:  
 - Template Studio Wizard Version: 
 - Windows Build:

**Comment by mikebattista** [comment]
Wizard and template version should be the same. They should be updated to 5.0. That should handle all user visible versions.

Assembly versions should be differentiated across extensions to avoid conflicts when running multiple extensions side by side.

**Comment by mikebattista** [comment]
Please update the C++ extension Product Name as well.

**Comment by mrlacey** [comment]
The version number specified in the vsixmanifest
![image](https://user-images.githubusercontent.com/189547/165507025-3a7db762-02b0-453d-b762-eab781635de1.png)

is what is also shown in the wizard

![image](https://user-images.githubusercontent.com/189547/165507082-60652c1d-4735-48a8-8f84-5dc307db65be.png)

and also in the TemplateStudio.xml (or appxmanifest)

![image](https://user-images.githubusercontent.com/189547/165507178-d8321e5b-32c1-4795-a042-3c9a7828fffb.png)

----

There is also now no separate Template version number (just the Wizard version number) as they're always distributed together.

**Comment by mrlacey** [comment]
> Assembly versions should be differentiated across extensions to avoid conflicts when running multiple extensions side by side.

For future clarification (if anyone else comes across this): 
While originally migrating from a single extension to multiple extensions, I enountered some issues where VS was trying to load an assembly from a different version of the extension that was also installed. Using different version numbers was one of the things I used to ensure that the assemblies were definitely different. It may be that other changes have made this unnecessary but changing this now is unnecessary. (Everything works and changing the version numbers so they're all the same would bring no benefit but risks breaking things.)
When debugging "weird" issues about assembly resolution and paths, it's useful to have different version numbers as this has helped highlight where things are wrong and where the wrong reference is coming from. The code related to this is likley to never need to be changed again (now it's working) but if it does, it could be useful to have these different version numbers there to help if changes introduce issues.

This does not apply to the version numbers specified in the appxmanifest and are seen everywhere.
